osquery/tables/system/linux/shadow.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <shadow.h>
+
+#include <boost/regex.hpp>
+
+#include <mutex>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+Mutex spwdEnumerationMutex;
```

Comment:
Does this need to be at such a high scope? Could it be brought down into your `genShadow` function where it's used? We try to avoid having variables live at this high of scope that aren't `const`.